### ISSUE TITLE
Provider doesn't allow for empty string in `parentResourcePath` section of resource identifier.

### ISSUE THREAD

**Issue by ringods** [pulumi/pulumi-azure-native#2516]
### What happened?

Customer tried to import a resource of type `azure-native:authorization:ManagementLockAtResourceLevel`:

```sh
$ pulumi import azure-native:authorization:ManagementLockAtResourceLevel DoNotDelete \
/subscriptions/32b9cb2e-69be-4040-80a6-02cd6b2cc5ec/resourceGroups/somerg1c68fe1c/providers/Microsoft.Network//virtualNetworks/eonnet9a968ea0/providers/Microsoft.Authorization/locks/DoNotDelete

...

Diagnostics:
  azure-native:authorization:ManagementLockAtResourceLevel (DoNotDelete):
    error: Preview failed: failed to parse '/subscriptions/32b9cb2e-69be-4040-80a6-02cd6b2cc5ec/resourceGroups/somerg1c68fe1c/providers/Microsoft.Network/virtualNetworks/somenet9a968ea0/providers/Microsoft.Authorization/locks/DoNotDelete' against the path '/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{parentResourcePath}/{resourceType}/{resourceName}/providers/Microsoft.Authorization/locks/{lockName}'
```

### Expected Behavior

The import command should work.

### Steps to reproduce

Deploy the following program on Azure:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as resources from "@pulumi/azure-native/resources";
import * as network from "@pulumi/azure-native/network";

// Create an Azure Resource Group
const resourceGroup = new resources.ResourceGroup("eonrg");

// Create an Azure resource (Virtual Network)
const eonnet = new network.VirtualNetwork("eonnet", {
    resourceGroupName: resourceGroup.name,
    addressSpace: {
        addressPrefixes: ["10.0.0.0/16"],
    },
    location: "WestEurope",
});
```

Once deployed, go the virtual network in the Azure Portal and add a `Delete` lock to the resource:

<img width="1533" alt="Screenshot 2023-06-13 at 18 26 52" src="https://github.com/pulumi/pulumi-azure-native/assets/77923/a9fda913-712c-4767-ab2c-b177b35a22e6">

Import the lock into the Pulumi program. Contrary to the documentation, I added an additional `/` between `Microsoft.Network` (`resourceProviderNamespace`) and `virtualNetworks` (`resourceType`). This should make `parentResourcePath` an empty string.

```sh
$ pulumi import azure-native:authorization:ManagementLockAtResourceLevel DoNotDelete \
/subscriptions/32b9cb2e-69be-4040-80a6-02cd6b2cc5ec/resourceGroups/somerg1c68fe1c/providers/Microsoft.Network//virtualNetworks/eonnet9a968ea0/providers/Microsoft.Authorization/locks/DoNotDelete
```

### Output of `pulumi about`

CLI          
Version      3.71.0
Go Version   go1.20.4
Go Compiler  gc

Plugins
NAME          VERSION
azure-native  1.103.0
nodejs        unknown

Host     
OS       darwin
Version  13.3.1
Arch     arm64

This project is written in nodejs: executable='/Users/ringods/.volta/bin/node' version='v16.19.1'

Current Stack: team-ce/import-locks/ringo

TYPE                                  URN
pulumi:pulumi:Stack                   urn:pulumi:ringo::import-locks::pulumi:pulumi:Stack::import-locks-ringo
pulumi:providers:azure-native         urn:pulumi:ringo::import-locks::pulumi:providers:azure-native::default_1_103_0
azure-native:resources:ResourceGroup  urn:pulumi:ringo::import-locks::azure-native:resources:ResourceGroup::eonrg
azure-native:network:VirtualNetwork   urn:pulumi:ringo::import-locks::azure-native:network:VirtualNetwork::eonnet

Found no pending operations associated with team-ce/ringo

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/ringods
User           ringods
Organizations  ringods, team-ce, pulumiverse, demo

Dependencies:
NAME                  VERSION
@pulumi/azure-native  1.103.0
@pulumi/pulumi        3.70.0
@types/node           16.18.34

## Current Code

The Azure API Path to manipulate is this:

`"/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{parentResourcePath}/{resourceType}/{resourceName}/providers/Microsoft.Authorization/locks/{lockName}"`

Although the `{parentResourcePath}` is a mandatory section, an [empty string is allowed](https://github.com/Azure/azure-sdk-for-go/issues/440#issuecomment-258312373):

> The `parentResourcePath` parameter is to be used when you need some extra stuff in the path... For example SQL Databases. For other resources you can just pass an empty string without problems.

This is the current code mapping an Azure API path to the regex matching an actual resource id:

https://github.com/pulumi/pulumi-azure-native/blob/60c55f5eabe3b6566359c270af61fca80d84c7b5/provider/pkg/resources/convert.go#L347-L358

This code results in the following regex being generated for `azure-native:authorization:ManagementLockAtResourceLevel`:

`(?i)^/subscriptions/(?P<subscriptionId>.+)/resourcegroups/(?P<resourceGroupName>.+)/providers/(?P<resourceProviderNamespace>.+)/(?P<parentResourcePath>.+)/(?P<resourceType>.+)/(?P<resourceName>.+)/providers/Microsoft.Authorization/locks/(?P<lockName>.+)$`

Matching an actual value against this regex doesn't work: https://regex101.com/r/PfRMI3/1

If I change the regex section for `parentResourcePath` from `(?P<parentResourcePath>.+)` to `(?P<parentResourcePath>.*?)`, it matches an empty string correctly: https://regex101.com/r/HTqAtO/1

Is it possible to give the `parentResourcePath` a special treatment?

## Local experiment

I patched the provider and replaced all the parts for the regex from:

https://github.com/pulumi/pulumi-azure-native/blob/60c55f5eabe3b6566359c270af61fca80d84c7b5/provider/pkg/resources/convert.go#L352

to

```go
regexParts[i] = fmt.Sprintf("(?P<%s>.*?)", name)
```

Setting this for all parts weakens the verification, so this is not a production ready patch. Now with this in place, I could import the lock correctly:

<img width="1980" alt="Screenshot 2023-06-13 at 18 02 57" src="https://github.com/pulumi/pulumi-azure-native/assets/77923/a1818018-ed33-45df-b917-1c5b71334b59">

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
If I've understood the fix correctly - that you're just changing the regex quantifier from `+` to `*` to also allow zero matche - I think this is a perfectly reasonable change.
